addon/components/base/bs-carousel.js

```diff
@@ -505,6 +514,9 @@ export default class Carousel extends Component.extend(ComponentParent) {
     } else {
       this.get('cycle').perform();
     }
+    if (this.get('onSlideChanged')) {
```

Comment:
As you have `onSlideChanged` defined on the component as a noop method, `this.get('onSlideChanged')` will always be truthy, so you could remove that guard and call the action directly.